examples/edge_intelligence/windmill_sklearn_edge.py

```diff
@@ -0,0 +1,146 @@
+#  modification, are permitted provided that the following conditions are met:#
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+# This example depicts how edge component can be used to take actions locally based on sklearn model
+# and actions can then be send to device using actuator_udm method defined here, which is currently just printing
+# actions on console
+
+import pint
+import math
+import Queue
+
+from liota.dccs.graphite import Graphite
+from liota.entities.metrics.metric import Metric
+from liota.entities.devices.sensor_tag import Sensors, SensorTagCollector
+from liota.entities.edge_systems.dell5k_edge_system import Dell5KEdgeSystem
+from liota.dcc_comms.socket_comms import SocketDccComms
+from liota.dccs.dcc import RegistrationFailure
+from liota.edge_component.sklearn_edge_component import SKLearnEdgeComponent
+
+# getting values from conf file
+config = {}
+execfile('../sampleProp.conf', config)
+
+# create a pint unit registry
+ureg = pint.UnitRegistry()
+
+rpm_model_queue = Queue.Queue()
+
+def get_rpm(sensor_tag_collector):
+	# RPM of Z-axis
+	# Average of 5 samples
+	_rpm_list = []
+	while True:
+		if len(_rpm_list) == 5:
+			rpm = 0
+			for _ in _rpm_list:
+				rpm += _
+			rpm = int(rpm / 5)
+			break
+		else:
+			z_degree = sensor_tag_collector.get_gyroscope()[2]
+			# (°/s to RPM)
+			_rpm_list.append(int((abs(z_degree) * 0.16667)))
+	rpm_model_queue.put(rpm)
+	return rpm
+
```

Comment:
Explain the queue approach for better understanding.

Reply:
Explained.